enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/RaftMachine.java

```diff
@@ -280,6 +308,24 @@ public synchronized ConsensusOutcome handle( RaftMessages.RaftMessage incomingMe
         return outcome;
     }
 
+    private void logDelay( RaftMessages.RaftMessage message )
+    {
+        long delay = Duration.between( Instant.now( clock ), message.receivedAt() ).toMillis();
+
+        raftMessageDelayMonitor.setDelay( delay );
+
+        long heartbeatInterval = leaderAvailabilityTimers.getHeartbeatInterval();
+        long electionTimeout = leaderAvailabilityTimers.getElectionTimeout();
+        if ( delay >= electionTimeout )
+        {
+            log.warn( "Message %s processing delay %s greater than election timeout interval %s", message, delay, electionTimeout );
```

Comment:
I am skeptical to this warning, which could have a great spamming and unnecessary support-inducing potential. Having the metrics available should perhaps be a sufficient solution?